### ISSUE TITLE
Cannot install luarocks 2.4.x

### ISSUE THREAD

**Issue by waruqi** [leafo/gh-actions-luarocks#8]
It works fine when installing luarocks 3.x, but cannot install luarocks 2.x

```
      - uses: leafo/gh-actions-lua@v8
      - uses: leafo/gh-actions-luarocks@v4
        with:
          luarocksVersion: 2.4.3
```

```console
/bin/tar xz --warning=no-unknown-keyword -C .install -f /home/runner/work/_temp/d4cf169e-ed6e-4081-a089-0be2837739e3
/home/runner/work/luarocks-build-xmake/luarocks-build-xmake/.install/luarocks-2.4.3/configure --with-lua-bin=/home/runner/work/luarocks-build-xmake/luarocks-build-xmake/.lua/bin --prefix=/home/runner/work/luarocks-build-xmake/luarocks-build-xmake/.luarocks
Lua interpreter not found in /home/runner/work/luarocks-build-xmake/luarocks-build-xmake/.lua/bin
You may want to use the flags --with-lua, --with-lua-bin and/or --lua-suffix. See --help.

configure failed.
```

https://github.com/xmake-io/luarocks-build-xmake/runs/1715427126?check_suite_focus=true

**Comment by alerque** [comment]
I just ran into this too, 3.x works fire but anything 2.x fails.

Since ferreting out Luarocks 2.x (hello all Ubuntu LTS releases) issues with my rockspecs is the reason I was trying to setup CI this time around at all, this is kind of a setback :(

**Comment by alerque** [comment]
Also note the combination of luajit / luajit-openresty and Luarocks 3.7.0 doesn't go so well. 3.1.3 is okay.

**Comment by leafo** [comment]
The reason is that you are using a version of Lua too new for LuaRocks. The Lua action by default installs Lua 5.4, and it is not compatible with the configure script for LuaRocks 2.4. Use an older version of Lua like 5.3 if you wish to use LuaRocks 2.4.

**Comment by alerque** [comment]
I see two issues with this closure:

1) It isn't just new Lua's that are a problem. If you look at my build sample for example it is failing with Lua 5.1, 5.2, luajit, etc.

2) I also recommend using modern versions of LuaRocks, but if the whole point of testing is making sure your library project works with ... say ... the version of LuaRocks shipped in the latest Ubuntu LTS release, then what's a fellow to do? Ruling out Lua 5.4 paired with LuaRocks 2.x is one thing, but the issue here is that 2.x won't install at all no matter what Lua you use.

**Comment by leafo** [comment]
@alerque  I was replying to the original poster and the error you shared in your screenshot:

> Lua interpreter not found in /home/runner/work/luarocks-build-xmake/luarocks-build-xmake/.lua/bin
You may want to use the flags --with-lua, --with-lua-bin and/or --lua-suffix. See --help.

This error is due to using a version of Lua that is too new for the version of LuaRocks you're using. Feel free to open a new issue with a more detailed error for the combination you're using.

**Comment by alerque** [comment]
I'm sorry but I think you're missing the point, and I don't see a good reason to start a new issue to report exactly the same problem. "Too old a version" is not the same as "interpreter not found". The issue is not that Lua is too new. That may well be true for Lua 5.4, but it doesn't work with Lua 5.1 either!

I setup a demo of this failing here:

https://github.com/lunarmodules/moon-sand/actions/runs/2042349859/workflow

I expect the 5.4 + 2.4.2 pairing to fail, but note that **all** of the 2.4.2 ones are failing, even the ones using OLD Luas.

This is important because (very sadly) the version of LuaRocks shipped with Ubuntu 20.04 is still 2.4.2. I hate it but *many many projects* are kind of stuck supporting it for awhile yet.

**Comment by leafo** [comment]
They are two separate issues. The original poster included the following code workflow:

      - uses: leafo/gh-actions-lua@v8
      - uses: leafo/gh-actions-luarocks@v4
        with:
          luarocksVersion: 2.4.3

And included the specific error message that happens when you use this workflow. (`Lua interpreter not found in ...`)

This workflow will never work and will always fail with the error message. I indicated that in my response and closed their issue.

The problem you're having is from a different workflow (eg. 2.4.2 + 5.2), and it produces a completely different error message, and it is caused by another problem. (`lua: ...e/runner/work/moon-sand/moon-sand/.luarocks/bin/luarocks:4: module 'luarocks.cfg' not found:`)

Regardless, it is no longer necessary to create a new issue for it as it has been fixed in this commit: https://github.com/leafo/gh-actions-luarocks/commit/eab7d674e82a0b1516ae6d24cc4ca329ff8e320f

**Comment by alerque** [comment]
You are correct those are different issues, I was looking at the wrong place for errors on my own workflows.

Thanks for the fix.

**Comment by leafo** [comment]
@alerque https://github.com/leafo/gh-actions-luarocks/releases/tag/v4.3.0 is now out with the fix so you can test it on your build

**Comment by alerque** [comment]
Confirmed. For anyone interested in how to have a matrix with multiple Luas and LuaRocks but exclude some, here is an example run demonstrating that this bug is fixed and how to avoid the 5.4/2.x pairing:

https://github.com/lunarmodules/moon-sand/actions/runs/2042501772/workflow
